projects/batfish-client/src/org/batfish/client/Client.java

```diff
@@ -315,6 +325,176 @@ private boolean answer(String questionTemplateName, String paramsLine,
+            }
+            // TODO: Need to double check isInetAddress()
+            if (!InetAddresses.isInetAddress(value.textValue())) {
+               throw new BatfishException(
+                     String.format("The input: %s is not a valid ip address.",
+                           value.textValue()));
+            }
+            break;
+         case IP_WILDCARD:
+            if (!value.isTextual()) {
+               throw new BatfishException(errorMessage);
+            }
+            // TODO: need ip_WildCard validation check
+            break;
+         case JAVA_REGEX:
+            if (!value.isTextual()) {
+               throw new BatfishException(errorMessage);
+            }
+            try {
+               Pattern.compile(value.textValue());
+            } catch (PatternSyntaxException e) {
+               throw new BatfishException(
+                     String.format("Input String: %s is not a valid Java " +
+                           "regular  expression for variable type: %s.",
+                           value.textValue(), expectedType));
+            }
+            break;
+         case JSON_PATH_REGEX:
+            if (!value.isTextual()) {
+               throw new BatfishException(errorMessage);
+            }
+            validateJsonPathRegex(value.textValue());
+            break;
+         case PREFIX:
+            if (!value.isTextual()) {
+               throw new BatfishException(errorMessage);
+            }
+            // TODO: need prefix validation check
+            break;
+         case PREFIX_RANGE:
+            if (!value.isTextual()) {
+               throw new BatfishException(errorMessage);
+            }
+            // TODO: need prefix range validation check
+            break;
+         case STRING:
+            if (!value.isTextual()) {
+               throw new BatfishException(errorMessage);
+            }
+            break;
+         case SUBRANGE:
+            if (!value.isTextual()) {
+               throw new BatfishException(errorMessage);
+            }
+            // TODO: need prefix range validation check
+            break;
+         case JSON_PATH:
+            validateJsonPath(value);
+            break;
+      }
+   }
+
+   /**
+    * Validate that {@code jsonPathRegex} contains a valid Java regular
+    * expression of a JsonPath(Starts with '/', ends with either '/' or
+    * '/i', contains a valid Java regular expression between '/').
+    * @throws BatfishException If the content of {@code jsonPathRegex} is not
+    * a valid java regular expression of a JsonPath.
+    */
+   static void validateJsonPathRegex(String jsonPathRegex)
+         throws BatfishException {
+      if (jsonPathRegex == null || jsonPathRegex.isEmpty()) {
+         throw new BatfishException(
+               "JsonPathRegex should not be empty or null.");
+      }
+      // TODO: Currently assume "/" and "/i" is allowed here.
+      if (jsonPathRegex.startsWith("/")) {
```

Comment:
This sequence of `if`s would be a lot cleaner and less nested using the following pattern:

```java
if (BAD CONDITION) {
   throw exception;
}

if (OTHER BAD CONDITION) {
   throw exception;
}

... good code ...
```

You have much less nesting if you "exit early" instead of having

```java
if (GOOD CONDITION) {
   if (OTHER GOOD CONDITION) {
      ... good code ...
   } else {
      throw exception
   }
} else {
   throw exception;
}
```